x-pack/plugins/observability_solution/logs_explorer/public/customizations/logs_explorer_profile.tsx

```diff
@@ -99,6 +109,51 @@ export const createLogsExplorerProfileCustomizations =
       },
     });
 
+    const isClickTriggerEvent = (
+      e: CustomClickTriggerEvent
+    ): e is WithPreventableEvent<ClickTriggerEvent['data']> => {
+      return Array.isArray(e.data) && 'column' in e.data[0];
+    };
+
+    const isMultiValueTriggerEvent = (
+      e: CustomClickTriggerEvent
+    ): e is WithPreventableEvent<MultiClickTriggerEvent['data']> => {
+      return Array.isArray(e.data) && 'cells' in e.data[0];
+    };
+
+    // Fix bug where filtering on histogram does not work
+    customizations.set({
+      id: 'unified_histogram',
+      onFilter: async (eventData) => {
+        eventData.preventDefault();
+        let filters;
+        if (isClickTriggerEvent(eventData)) {
+          filters = await discoverServices.data.actions.createFiltersFromValueClickAction(
+            eventData
+          );
+        } else if (isMultiValueTriggerEvent(eventData)) {
+          filters = await discoverServices.data.actions.createFiltersFromMultiValueClickAction(
+            eventData
+          );
+        }
+        if (filters && filters.length > 0) {
+          data.query.filterManager.addFilters(filters);
+        }
+      },
+      onBrushEnd: (eventData) => {
+        eventData.preventDefault();
+
+        const [from, to] = eventData.range;
+        discoverServices.data.query.timefilter.timefilter.setTime({
+          from: new Date(from).toISOString(),
+          to: new Date(to).toISOString(),
+          mode: 'absolute',
+        });
+      },
```

Comment:
**nit**: would be great if you could move this customization implementation to a dedicated file that exports these handlers, just to keep the logs_explorer_profile the entry point for applying the customization and keep it the most readable. 

A `custom_unified_histogram` module could be good for keeping the naming consistent with the other modules, wdyt?

Reply:
Sounds good, I've refactored the histogram into it's own file